gluten-core/src/main/scala/org/apache/gluten/execution/DataSourceScanTransformerRegister.scala

```diff
@@ -30,13 +30,13 @@ trait DataSourceScanTransformerRegister {
   /**
    * The class name that used to identify what kind of datasource this is。
    *
-   * For DataSource V1, it should be the child class name of
-   * [[org.apache.spark.sql.execution.datasources.FileIndex]].
+   * For DataSource V1, it should be relation.fileFormat like
```

Comment:
@YannByron For `org.apache.spark.sql.execution.datasources.FileIndex`, it can be used to distinguish different datasources but it's too general that all kinds of files read will pass, such as meta data/log files used for query plan analysis. It is not necessary and may trigger failures in some corner cases. So here, we limit it to the parquet format, are you okay for this change?

Reply:
`scanClassName` is a flag to decide which data source transformer should be triggered during register, such as delta, iceberg, hudi, etc. It's used to be determined by value of `org.apache.spark.sql.execution.datasources.FileIndex`, now in this PR we change to `relation.fileFormat`.